plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -512,4 +503,15 @@ private static TableIdentifier toIdentifier(SchemaTableName schemaTableName)
     {
         return TableIdentifier.of(schemaTableName.getSchemaName(), schemaTableName.getTableName());
     }
+
+    private List<Namespace> listNamespaces(ConnectorSession session, Optional<String> namespace)
+    {
+        if (namespace.isEmpty()) {
+            return listNamespaces(session).stream()
+                    .map(Namespace::of)
+                    .collect(toImmutableList());
+        }
+
+        return namespaceExists(session, namespace.get()) ? ImmutableList.of(Namespace.of(namespace.get())) : ImmutableList.of();
```

Comment:
@findepi on this comment https://github.com/trinodb/trino/pull/19818#discussion_r1409263532

The distinction has to do with any permissions on the catalog server side. I think the issue is if an explicit namespace is provided we ideally should make the most granular request (GetNamespaceRequest) since they may be allowed to do that for the specific namespace. Whereas they may not be allowed to list all the namespaces (thus a list + filter may fail for the user at the time of listing). Admittedly this does depend on the catalog server side implementation but I think it's a reasonable behavior which generalizes.  

A secondary benefit is that if an explicit namespace is provided and GetNamespace request is made it avoids a heavier listing on the server side. Ultimately, I'd advocate for not changing this behavior.

Reply:
Ah sorry for the confusion, I thought you were implying that we should keep the list call and then filter for the specific namespace. I think what you are saying is the existence check simply isn't needed. We can just use the passed in value, and during `listTables` if the namespace is concurrently removed it just wouldn't get surfaced. 

I removed the existence check.